pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,554 @@
+						"name": p.Name,
+						"kind": p.Kind,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: p.Name,
+							Env: []corev1.EnvVar{
+								{Name: "PORT", Value: fmt.Sprintf("%d", p.Port)},
+							},
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: int32(p.Port)},
+							},
+							Image:           p.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         p.Command,
+						},
+					},
+					Affinity: p.Affinity,
+				},
+			},
+			Replicas: &replicas32,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": p.Name,
+					"kind": p.Kind,
+				},
+			},
+		},
+	}
+}
+
+type k8sConnectivityParams struct {
+	clients         *deploymentClients
+	client          k8sClientOperations
+	params          Parameters
+	antreaNamespace string
+	clientPods      *corev1.PodList
+	echoPods        map[string]string
+}
+
+func NewAntreaConnectivityCheck(client k8sClientOperations, p Parameters) *k8sConnectivityParams {
+	return &k8sConnectivityParams{
+		client:          client,
+		antreaNamespace: "kube-system",
+		params:          p,
+	}
+}
+
+func (k *k8sConnectivityParams) validatePodToPod(ctx context.Context) {
+	//conducts Pod to Pod connectivity tests within same node and different node
+	for _, clientPod := range k.clientPods.Items {
+		for echoName, echoIP := range k.echoPods {
+			var (
+				srcPod  = connectivityCheckNamespace + "/" + clientPod.Name
+				dstPod  = connectivityCheckNamespace + "/" + echoName
+				success = true
+			)
+			k.Header("Validating from pod %s to pod %s...", srcPod, dstPod)
+			_, err := k.client.ExecInPod(ctx, connectivityCheckNamespace, clientPod.Name, "", curlCommand(echoIP+":80"))
+			if err != nil {
+				k.Log("curl connectivity check command failed: %s", err)
+				success = false
+			}
+
+			if success {
+				k.Log("client pod %s was able to communicate with echo pod %s (%s)", clientPod.Name, echoName, echoIP)
+			} else {
+				k.Log("client pod %s was not able to communicate with echo pod %s (%s)", clientPod.Name, echoName, echoIP)
+			}
+
+			k.Relax()
+		}
+	}
+}
+
+func (k *k8sConnectivityParams) validatePodInternetConnectivity(ctx context.Context) error {
+	for _, clientPod := range k.clientPods.Items {
+		var (
+			srcPod  = connectivityCheckNamespace + "/" + clientPod.Name
+			success = true
+		)
+
+		k.Header("Validating connectivity from pod %s to the world (google.com)...", srcPod)
+		_, err := k.client.ExecInPod(ctx, connectivityCheckNamespace, clientPod.Name, clientDeploymentName, curlCommandforInternet("https://google.com"))
+		if err != nil {
+			k.Log("Connectivity test from pod %s to google.com failed: %s", srcPod, err)
+			success = false
+		}
+
+		if success {
+			k.Log("Pod %s was able to connect to google.com", srcPod)
+		}
+
+		k.Relax()
+	}
+	return nil
+}
+
+func (k *k8sConnectivityParams) Relax() {
+	time.Sleep(2 * time.Second)
+}
+
+type Parameters struct {
```

Comment:
I do not understand how these parameters are set? I didn't see any corresponding command-line flags?

Reply:
I have added corresponding command-line flags now :)